dbt/adapters/athena/impl.py

```diff
@@ -253,15 +252,43 @@ def _get_one_catalog(
         schemas: Dict[str, Optional[Set[str]]],
         manifest: Manifest,
     ) -> agate.Table:
-        kwargs = {"information_schema": information_schema, "schemas": schemas}
-        table = self.execute_macro(
-            GET_CATALOG_MACRO_NAME,
-            kwargs=kwargs,
-            # pass in the full manifest so we get any local project
-            # overrides
-            manifest=manifest,
-        )
+        conn = self.connections.get_thread_connection()
+        client = conn.handle
 
+        with boto3_client_lock:
+            glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
+
+        catalog = []
+        paginator = glue_client.get_paginator("get_tables")
+        for schema, relations in schemas.items():
+            for page in paginator.paginate(DatabaseName=schema, MaxResults=100):
+                for table in page["TableList"]:
+                    if table["Name"] in relations:
+                        table_catalog = {
+                            "table_database": conn.credentials.database,
+                            "table_schema": table["DatabaseName"],
+                            "table_name": table["Name"],
+                            "table_type": self.relation_type_map[table["TableType"]],
+                            "table_comment": table.get("Parameters", {}).get("comment", table.get("Description", "")),
+                        }
+                        catalog.extend(
+                            [
+                                {
+                                    **table_catalog,
+                                    **{
+                                        "column_name": col["Name"],
+                                        "column_index": idx,
+                                        "column_type": col["Type"],
+                                        "column_comment": col.get("Comment", ""),
+                                    },
+                                }
+                                for idx, col in enumerate(
+                                    table["StorageDescriptor"]["Columns"] + table.get("PartitionKeys", [])
+                                )
+                            ]
```

Comment:
just a thought, shall we move this piece in a little private function? to make the code a bit more readable. also the private function could be unit tested separated.

Reply:
Done. Right now I did not do any unit testing on this specific function since this is already covered but if you want I can add one.